scte35/doc.go

```diff
@@ -151,12 +137,40 @@ type SegmentationDescriptor interface {
 	UPIDType() SegUPIDType
 	// UPID returns the upid of the descriptor
 	UPID() []byte
-	// Compare returns results in terms of trumping rules, <0 if sd is less than, 0
-	// if equal, and >0 if greater
-	Compare(sd SegmentationDescriptor) int
 	// CanClose returns true if this descriptor can close the passed in descriptor
 	CanClose(out SegmentationDescriptor) bool
 	// Equal returns true/false if segmentation descriptor is functionally
 	// equal (i.e. a duplicate)
 	Equal(sd SegmentationDescriptor) bool
 }
+
+// State maintains current state for all signals and descriptors.  The intended
+// usage is to call ParseSCTE35() on raw data to create a signal, and then call
+// ProcessSignal with that signal.  This returns the list of descriptors closed
+// by that signal. If signals have a duration and need to be closed implicitly
+// after some timer has passed, then Close() can be used for that.  Some
+// example code is below.
+// s := scte35.NewState()
+// scte,_ := scte.ParseSCTE35(bytes)
+// for _,d := range(scte.Descriptors()) {
+//   closed = s.ProcessDescriptor(d)
+//   ...handle closed signals appropriately here
+//   if d.HasDuration() {
+//     time.AfterFunc(d.Duration() + someFudgeDelta,
+//                    func() { closed = s.Close(d) })
+//   }
+// }
+type State interface {
+	// Open returns a list of open signals
+	Open() []SegmentationDescriptor
+	// Process takes a scte35 descriptor and returns a list of descriptors closed by it
+	ProcessDescriptor(desc SegmentationDescriptor) ([]SegmentationDescriptor, error)
```

Comment:
SegmentationDescriptor is an interface which is to pointers (one to type, one to data).  We can't change this to a reference, because golang will not allow it.
Ref: http://stackoverflow.com/questions/13511203/why-cant-i-assign-a-struct-to-an-interface